### ISSUE TITLE
Promoting Optimistic Provide from experimental to production ready

### ISSUE THREAD

**Issue by dennis-tra** [ipfs/kubo#10047]
### Checklist

- [X] My issue is specific & actionable.
- [X] I am not suggesting a protocol enhancement.
- [X] I have searched on the [issue tracker](https://github.com/ipfs/kubo/issues?q=is%3Aissue) for my issue.

### Description

# Context 

With Kubo 0.20.0 we released the experimental flag to [enable Optimistic Provide](https://github.com/ipfs/kubo/blob/master/docs/experimental-features.md#optimistic-provide). A technique that can dramatically accelerate DHT PUT performance. Recently, we measured sub-second provide latencies from North America and central Europe for >90% percent of the provide operations that we performed (`N > 10k`) while reducing networking overhead by >40% without compromising record availability.

**This is a tracking issue to discuss when and if we should promote it to a production-ready feature.**

# Outreach/Adoption

I reached out to a couple of projects to gather feedback:

- **NuNet:**
  - _PR:_https://gitlab.com/nunet/device-management-service/-/issues/215
  - _Status:_ blocked on [another issue](https://github.com/libp2p/go-libp2p-kad-dht/issues/848) in go-libp2p-kad-dht but has signalled interest
- **vocdoni:**
  - _PR:_ https://github.com/vocdoni/vocdoni-node/issues/989
  - _Status:_ plans to enable the option in [this PR](https://github.com/vocdoni/vocdoni-node/issues/989) but want to have monitoring in place before merging
- **Mintter:**
  - _PR:_ https://github.com/mintterteam/mintter/issues/1397
  - _Status:_ n/a
- **pocket:**
  - _PR:_ https://github.com/pokt-network/pocket/issues/819
  - _Status:_ signaled interest but also stumbled upon other issues
- **WeAreAmaze:**
  - _PR:_ https://github.com/WeAreAmaze/amc/issues/40
  - _Status:_ n/a
- **Shutter:**
  - _PR:_ https://github.com/shutter-network/rolling-shutter/issues/378
  - _Status:_ n/a
- **Indra:**
  - _PR:_ https://github.com/indra-labs/indra/issues/4
  - _Status:_ @l0k18 from [indra-labs](https://github.com/indra-labs) has given [promising feedback](https://github.com/indra-labs/indra/issues/4#issuecomment-1645180778).
- **kubeedge:**
  - _PR:_ https://github.com/kubeedge/kubeedge/discussions/4820
  - _Status:_ n/a
- **dymint:**
  - _PR:_ https://github.com/dymensionxyz/dymint/issues/350
  - _Status:_ signaled interest
- **b7s:**
  - _PR:_ https://github.com/blocklessnetwork/b7s/issues/90
  - _Status:_ signaled interest

# Production-Ready Criteria

TBD

# Kubo

At the moment, the feature is a global flag that enables the technique for every provide operation. Given the specific trade-offs of the mechanism, I would rather apply it for specific use-cases only.

For example, non-time-sensitive operations like re-provides don't necessarily need to use it. On the other hand, if users add files to Kubo, I'd argue that they will expect them to be _immediately_ available in the network.

**Comment by Jorropo** [comment]
If this is production ready, shouldn't this change be made in go-libp2p-kad-dht ?

**Comment by lidel** [comment]
Closing: optimistic provide is enabled by default since v0.39 as part of the sweep provider work
